backend/wmg/pipeline/summary_cubes/builder.py

```diff
@@ -0,0 +1,435 @@
+
+from backend.wmg.data.constants import (
+    GENE_EXPRESSION_COUNT_MIN_THRESHOLD,
+    RANKIT_RAW_EXPR_COUNT_FILTERING_MIN_THRESHOLD,
+)
+from backend.wmg.data.rankit import rankit
+from backend.wmg.data.schemas.cube_schema import (
+    expression_summary_indexed_dims_no_gene_ontology,
+    expression_summary_non_indexed_dims,
+    expression_summary_schema,
+)
+from backend.wmg.data.schemas.cube_schema_default import (
+    expression_summary_indexed_dims as expression_summary_indexed_dims_default,
+)
+from backend.wmg.data.schemas.cube_schema_default import (
+    expression_summary_non_indexed_dims as expression_summary_non_indexed_dims_default,
+)
+from backend.wmg.data.schemas.cube_schema_default import expression_summary_schema as expression_summary_schema_default
+from backend.wmg.data.schemas.expression_summary_fmg_cube_schema import (
+    expression_summary_fmg_indexed_dims,
+    expression_summary_fmg_non_indexed_dims,
+    expression_summary_fmg_schema,
+)
+from backend.wmg.data.snapshot import (
+    EXPRESSION_SUMMARY_CUBE_NAME,
+    EXPRESSION_SUMMARY_DEFAULT_CUBE_NAME,
+    EXPRESSION_SUMMARY_FMG_CUBE_NAME,
+)
+from backend.wmg.data.tiledb import create_ctx
+from backend.wmg.data.utils import (
+    create_empty_cube,
+    log_func_runtime,
+    remove_accents,
+    return_dataset_dict_w_publications,
+)
+
+logger = logging.getLogger(__name__)
+
+DIMENSION_NAME_MAP_CENSUS_TO_WMG = {
+    "tissue_ontology_term_id": "tissue_original_ontology_term_id",
+    "tissue_general_ontology_term_id": "tissue_ontology_term_id",
+}
+
+
+@nb.njit(fastmath=True, error_model="numpy", parallel=False, nogil=True)
+def gene_expression_sum_x_cube_dimension(
+    rankit_values: np.ndarray,
+    obs_idxs: np.ndarray,
+    var_idx: np.ndarray,
+    cube_indices: np.ndarray,
+    sum_into: np.ndarray,
+    sqsum_into: np.ndarray,
+    nnz_into: np.ndarray,
+):
+    """
+    Sum the rankit values for each gene (for each cube row/combo of cell attributes)
+    Also track the number of cells that express that gene (nnz count)
+    """
+    for k in range(len(rankit_values)):
+        val = rankit_values[k]
+        if np.isfinite(val):
+            cidx = var_idx[k]
+            grp_idx = cube_indices[obs_idxs[k]]
+            sum_into[grp_idx, cidx] += val
+            sqsum_into[grp_idx, cidx] += val**2
+            nnz_into[grp_idx, cidx] += 1
+
+
+class SummaryCubesBuilder:
+    def __init__(self, *, corpus_path: str, organismInfo: dict):
+        """
+        Initialize the SummaryCubesBuilder class.
+
+        Args:
+            corpus_path (str): The path to the corpus.
+            organismInfo (dict): Information about the organism.
+                ex:
+                organismInfo = {
+                    "id": "UBERON:0000955",
+                    "label": "brain",
+                }
+        """
+        organism = organismInfo["label"]
+        organismId = organismInfo["id"]
```

Comment:
do you mean `organ` instead of `organism` here? The example of `brain` suggests to organ to me. Or perhaps the example is wrong it should be `homo sapiens` or something?

Reply:
Oh shoot, lol, nice catch! That's what I get for having copilot autocomplete my docstring.